### PR TITLE
[time-series] Fix back test dataloader

### PR DESCRIPTION
**Pull request by garethmd**

### CODE REVIEW

**Comment by kashif** [comment]
awesome thanks! having a look now

**Comment by kashif** [comment]
@garethmd  can we say something like: 

```

For training/validation/testing we next create an `InstanceSplitter` which is used to sample windows from the dataset (as, remember, we can't pass the entire history of values to the Transformer due to time- and memory constraints).

The instance splitter samples random `context_length` sized and subsequent `prediction_length` sized windows from the data, and appends a `past_` or `future_` key to any temporal keys in `time_series_fields` for the respective windows. The instance splitter can be configured into three different models:
1. `mode="train"`: Here we sample the context and prediction length windows randomly from the dataset given to it (the training dataset)
2. `model="validation": Here we sample the very last context length window and prediction window from the dataset given to it (for the back-testing or validation likelihood calculations)
3. `mode="test"`: Here we sample the very last context length window only (for the prediction use case)
```

**Comment by garethmd** [comment]
sure - but do you still want the test mode as it won't be used in the code?

**Comment by kashif** [comment]
yes as that is needed for the real-world prediciton use case... where you use the data you have

**Comment by garethmd** [comment]
@kashif I've updated the instance splitter text and put the test mode back in

**Comment by kashif** [comment]
yup! i am fixing up the notebook and test results... so i am thinking of using the comparison from the autogluon-timeseries paper which some better comparisons... the monash time series MASE table actually has a bug (in how they calculate the seasonalities)

**Comment by garethmd** [comment]
ah good to know that actually explains a lot as I couldn't reconcile the monash mase value with the hugging face one.

**Comment by kashif** [comment]
see the issue here from autogluon author: https://github.com/rakshitha123/TSForecasting/issues/14

File: autoformer.md [hunk 1]
```diff
@@ -440,7 +440,10 @@ def create_transformation(freq: str, config: PretrainedConfig) -> Transformation
...
 
-The instance splitter samples random `context_length` sized and subsequent `prediction_length` sized windows from the data, and appends a `past_` or `future_` key to any temporal keys for the respective windows. This makes sure that the `values` will be split into `past_values` and subsequent `future_values` keys, which will serve as the encoder and decoder inputs respectively. The same happens for any keys in the `time_series_fields` argument:
+The instance splitter samples random `context_length` sized and subsequent `prediction_length` sized windows from the data, and appends a `past_` or `future_` key to any temporal keys in `time_series_fields` for the respective windows. The instance splitter can be configured into three different models:
+1. `mode="train"`: Here we sample the context and prediction length windows randomly from the dataset given to it (the training dataset)
+2. `model="validation"`: Here we sample the very last context length window and prediction window from the dataset given to it (for the back-testing or validation likelihood calculations)
+3. `mode="test"`: Here we sample the very last context length window only (for the prediction use case)
```
Comment (pcuenca) [inline autoformer.md:446]:
```suggestion
The instance splitter samples random `context_length` sized and subsequent `prediction_length` sized windows from the data, and appends a `past_` or `future_` key to any temporal keys in `time_series_fields` for the respective windows. The instance splitter can be configured into three different modes:
1. `mode="train"`: Here we sample the context and prediction length windows randomly from the dataset given to it (the training dataset)
2. `mode="validation"`: Here we sample the very last context length window and prediction window from the dataset given to it (for the back-testing or validation likelihood calculations)
3. `mode="test"`: Here we sample the very last context length window only (for the prediction use case)
```

File: informer.md [hunk 2]
```diff
@@ -581,7 +581,10 @@ def create_transformation(freq: str, config: PretrainedConfig) -> Transformation
...
 
-The instance splitter samples random `context_length` sized and subsequent `prediction_length` sized windows from the data, and appends a `past_` or `future_` key to any temporal keys for the respective windows. This makes sure that the `values` will be split into `past_values` and subsequent `future_values` keys, which will serve as the encoder and decoder inputs respectively. The same happens for any keys in the `time_series_fields` argument:
+The instance splitter samples random `context_length` sized and subsequent `prediction_length` sized windows from the data, and appends a `past_` or `future_` key to any temporal keys in `time_series_fields` for the respective windows. The instance splitter can be configured into three different models:
+1. `mode="train"`: Here we sample the context and prediction length windows randomly from the dataset given to it (the training dataset)
+2. `model="validation"`: Here we sample the very last context length window and prediction window from the dataset given to it (for the back-testing or validation likelihood calculations)
+3. `mode="test"`: Here we sample the very last context length window only (for the prediction use case)
```
Comment (pcuenca) [inline informer.md:587]:
```suggestion
The instance splitter samples random `context_length` sized and subsequent `prediction_length` sized windows from the data, and appends a `past_` or `future_` key to any temporal keys in `time_series_fields` for the respective windows. The instance splitter can be configured into three different modes:
1. `mode="train"`: Here we sample the context and prediction length windows randomly from the dataset given to it (the training dataset)
2. `mode="validation"`: Here we sample the very last context length window and prediction window from the dataset given to it (for the back-testing or validation likelihood calculations)
3. `mode="test"`: Here we sample the very last context length window only (for the prediction use case)
```

File: time-series-transformers.md [hunk 3]
```diff
@@ -436,7 +436,10 @@ def create_transformation(freq: str, config: PretrainedConfig) -> Transformation
...
 
-The instance splitter samples random `context_length` sized and subsequent `prediction_length` sized windows from the data and appends a `past_` or `future_` key to any temporal keys for the respective windows. This makes sure that the `values` will be split into `past_values` and subsequent `future_values` keys, which will serve as the encoder and decoder inputs respectively. The same happens for any keys in the `time_series_fields` argument:
+The instance splitter samples random `context_length` sized and subsequent `prediction_length` sized windows from the data, and appends a `past_` or `future_` key to any temporal keys in `time_series_fields` for the respective windows. The instance splitter can be configured into three different models:
+1. `mode="train"`: Here we sample the context and prediction length windows randomly from the dataset given to it (the training dataset)
+2. `model="validation"`: Here we sample the very last context length window and prediction window from the dataset given to it (for the back-testing or validation likelihood calculations)
+3. `mode="test"`: Here we sample the very last context length window only (for the prediction use case)
```
Comment (pcuenca) [inline time-series-transformers.md:442]:
```suggestion
The instance splitter samples random `context_length` sized and subsequent `prediction_length` sized windows from the data, and appends a `past_` or `future_` key to any temporal keys in `time_series_fields` for the respective windows. The instance splitter can be configured into three different modes:
1. `mode="train"`: Here we sample the context and prediction length windows randomly from the dataset given to it (the training dataset)
2. `mode="validation"`: Here we sample the very last context length window and prediction window from the dataset given to it (for the back-testing or validation likelihood calculations)
3. `mode="test"`: Here we sample the very last context length window only (for the prediction use case)
```

File: time-series-transformers.md [hunk 4]
```diff
@@ -578,6 +581,46 @@ def create_test_dataloader(
     )
 ```
 
+We have a test dataloader helper for completion, even though we will not use it here:
```
Comment (pcuenca) [inline time-series-transformers.md:584]: Maybe copy the same comment in the others, I was wondering when we were using the `test` dataloader.

Comment (kashif) [reply time-series-transformers.md:584]: right let me add that... basically in "production" setting where one does not have a back-testing window

**Pull request merged**